dmsg-http.go

```diff
@@ -4,29 +4,24 @@ import (
 	"net/http"
 	"time"
 
-	"github.com/SkycoinProject/dmsg/cipher"
-	"github.com/SkycoinProject/dmsg/disc"
+	"github.com/SkycoinProject/dmsg"
 )
 
-// DefaultDMSGClient creates http Client using default discovery service
-// Default value can be found in dmsghttp.DefaultDiscoveryURL
-func DefaultDMSGClient(pubKey cipher.PubKey, secKey cipher.SecKey) *http.Client {
-	return DMSGClient(disc.NewHTTP(DefaultDiscoveryURL), pubKey, secKey)
-}
+const (
+	clientTimeout = 30 * time.Second
+)
 
-// DMSGClient creates http Client using provided discovery service and public / secret keypair
+// Client creates a http client
 // Returned client is using dmsg transport protocol instead of tcp for establishing connection
-func DMSGClient(discovery disc.APIClient, pubKey cipher.PubKey, secKey cipher.SecKey) *http.Client {
-	transport := DMSGTransport{
-		Discovery:  discovery,
-		PubKey:     pubKey,
-		SecKey:     secKey,
+func Client(dmsgC *dmsg.Client) *http.Client {
+	transport := Transport{
+		DmsgClient: dmsgC,
 		RetryCount: 20,
```

Comment:
Since refactor is in progress RetryCount should probably be moved next to clientTimeout

Reply:
I’m not sure if RetryCount is even needed. I feel like it can be removed.

Reply:
It was a requirement from Brandon when we started, that's why it's there